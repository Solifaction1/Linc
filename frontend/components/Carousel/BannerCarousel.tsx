import React from 'react'
import { Carousel } from 'react-responsive-carousel'

import Slide from './Slide/Slide'
import Arrow from './Arrow/Arrow'
import Indicator from './Indicator/Indicator'
import { propsConfig, defaultProps, Item } from './banner-carousel.config'

import styles from './banner-carousel.module.scss'
import 'react-responsive-carousel/lib/styles/carousel.min.css'

const BannerCarousel: React.FC<propsConfig> = (props: propsConfig) => {
	return (
		<div className={styles.carousel}>
			<Carousel
				{...props}
				renderArrowNext={Arrow('next', !!props.showArrows)}
				renderArrowPrev={Arrow('prev', !!props.showArrows)}
				renderIndicator={Indicator}
				showThumbs={false} // For testing
			>
				{props.items &&
					props.items.map((item: Item) => {
						return (
							<div key={item.title}>
								<Slide imgAlt={item.title} imgUrl={item.url} />
							</div>
						)
					})}
			</Carousel>
		</div>
	)
}

BannerCarousel.defaultProps = defaultProps

export default BannerCarousel
