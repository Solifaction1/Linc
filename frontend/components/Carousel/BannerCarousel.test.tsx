import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import BannerCarousel from './BannerCarousel'

describe('BannerCarousel', () => {
	it('renders a BannerCarousel', () => {
		const testItem = {
			title: 'title',
			url: 'url',
		}

		const { container } = render(<BannerCarousel items={[testItem]} />)

		expect(container).toMatchSnapshot()

		expect(container.getElementsByClassName('carousel')[0]).toBeVisible()
		expect(container.getElementsByClassName('carousel-root')[0]).toBeVisible()
		expect(
			container.getElementsByClassName('carousel carousel-slider')[0]
		).toBeVisible()
		expect(
			container.getElementsByClassName('indicator selected')[0]
		).toBeVisible()
		expect(container.getElementsByClassName('arrow prev')[0]).toBeVisible()
		expect(
			container.getElementsByClassName('slider-wrapper axis-horizontal')[0]
		).toBeVisible()
		expect(container.getElementsByClassName('slider animated')[0]).toBeVisible()
		expect(
			container.getElementsByClassName('slide selected previous')[0]
		).toBeVisible()
		expect(container.getElementsByClassName('slideContainer')[0]).toBeVisible()
	})
})
