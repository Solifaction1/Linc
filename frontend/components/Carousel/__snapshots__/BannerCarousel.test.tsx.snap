// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerCarousel renders a BannerCarousel 1`] = `
<div>
  <div
    class="carousel"
  >
    <div
      class="carousel-root"
      tabindex="0"
    >
      <div
        class="carousel carousel-slider"
        style="width: 100%;"
      >
        <ul
          class="control-dots"
        >
          <button
            aria-label="slide item 1"
            class="
				indicator
				selected
			"
            tabindex="0"
            type="button"
            value="0"
          />
        </ul>
        <button
          aria-label="prev"
          class="
					arrow
					prev
				"
          type="button"
        />
        <div
          class="slider-wrapper axis-horizontal"
        >
          <ul
            class="slider animated"
            style="transform: translate3d(-100%,0,0); transition-duration: 350ms;"
          >
            <li
              class="slide selected previous"
            >
              <div>
                <div
                  class="slideContainer"
                >
                  <img
                    alt="title"
                    src="url"
                  />
                </div>
              </div>
            </li>
            <li
              class="slide selected previous"
            >
              <div>
                <div
                  class="slideContainer"
                >
                  <img
                    alt="title"
                    src="url"
                  />
                </div>
              </div>
            </li>
          </ul>
        </div>
        <button
          aria-label="next"
          class="
					arrow
					next
				"
          type="button"
        />
      </div>
    </div>
  </div>
</div>
`;
